src/MongoDbOperationStore.ts

```diff
@@ -0,0 +1,139 @@
+    // Otherwise, we need to create the collection and define an index on
+    // (did, transactionNumber, operationIndex), first dropping the collection if
+    // it exists.
+    if (resuming) {
+      this.collection = this.db.collection(collectionsName);
+    } else {
+      await this.db.collection(collectionsName).drop();
+      this.collection = await this.db.createCollection(collectionsName, { strict: false });
+      await this.collection.createIndex({ didUniqueSuffix: 1, transactionNumber: 1, operationIndex: 1 });
+    }
+  }
+
+  /**
+   * Implement OperationStore.put.
+   */
+  public async put (operation: Operation): Promise<void> {
+    const mongoOperation = this.convertToMongoOperation(operation);
+    await this.collection!.insertOne(mongoOperation);
+  }
+
+  /**
+   * Implement OperationStore.putBatch
+   */
+  public async putBatch (operations: Array<Operation>): Promise<void> {
+    let batch = this.collection!.initializeUnorderedBulkOp();
+
+    for (const operation of operations) {
+      const mongoOperation = this.convertToMongoOperation(operation);
+      batch.insert(mongoOperation);
+    }
+
+    await batch.execute();
+  }
+
+  /**
+   * Get an iterator that returns all operations with a given
+   * didUniqueSuffix ordered by (transactionNumber, operationIndex)
+   * ascending.
+   */
+  public async get (didUniqueSuffix: string): Promise<Iterable<Operation>> {
+    const mongoOperations = await this.collection!.find({ didUniqueSuffix }).sort({ transactionNumber: 1, operationIndex: 1 }).toArray();
+    return mongoOperations.map(this.convertToOperation);
+  }
+
+  /**
+   * Delete all operations with transaction number greater than the
+   * provided parameter.
+   */
+  public async delete (transactionNumber?: number): Promise<void> {
+    if (transactionNumber) {
+      await this.collection!.deleteMany({ transactionNumber: { $gt: transactionNumber } });
+    } else {
+      await this.collection!.deleteMany({});
+    }
+  }
+
+  /**
+   * Convert a Sidetree operation to a more minimal MongoOperation object
+   * that can be stored on MongoDb. The MongoOperation object has sufficient
+   * information to reconstruct the original operation.
+   */
+  private convertToMongoOperation (operation: Operation): MongoOperation {
+    // Convert the operation buffer to string (mongo read-write not handled seamlessly for buffer types)
+    const operationBufferBase64Encoding = operation.operationBuffer.toString('base64');
+
+    return {
+      didUniqueSuffix: operation.getDidUniqueSuffix(),
+      operationBufferBase64: operationBufferBase64Encoding,
+      operationIndex: operation.operationIndex!,
+      transactionNumber: operation.transactionNumber!,
+      transactionTime: operation.transactionTime!,
+      batchFileHash: operation.batchFileHash!
+    };
+  }
+
+  /**
+   * Convert a MongoDB representation of an operation to a Sidetree operation.
+   * Inverse of getMongoOperation() method above.
```

Comment:
Should be `convertToMongoOperation` now.

Reply:
Good catch :-)